### PR TITLE
16846

### PR DESCRIPTION
**Pull request by wgreenberg**

### CODE REVIEW

File: test/lib/util.test.js [hunk 1]
```diff
@@ -104,6 +106,8 @@ describe('download_image', function() {
     });
 });
 
+// FIXME this text intermittently fails because it reaches out to an external
+// server. We should replace it with gzipped content served locally
```
Comment (sgnn7) [inline test/lib/util.test.js]: Fair enough - I didn't want to have copyrighted content in our repo and I didn't have a good way to simulate a gzipped server at the time which is why it's a link.

File: lib/index.js [hunk 2]
```diff
@@ -223,6 +223,9 @@ class Hatch {
             for (let outgoing_asset_id of asset.get_dependent_assets())
                 outgoing.add(outgoing_asset_id);
         });
+        this._videos.forEach((video) => {
```
Comment (sgnn7) [inline lib/index.js]: Isn't a video just another "dependent asset" like the list returned by `get_dependent_assets`? In other words, why treat this type differently from the rest?

Comment (wgreenberg) [inline lib/index.js]: yeah this bit's a tad confusing. Videos are dependent assets, and will be present in the `outgoing` set -- however, prior to this commit, they weren't considered part of the hatch's `asset_ids`, so this check failed

**Comment by sgnn7** [comment]
CR pass

**Pull request merged**